src/bin/download_sysext.rs

```diff
@@ -342,6 +378,28 @@ async fn main() -> Result<(), Box<dyn Error>> {
     fs::create_dir_all(&unverified_dir)?;
     fs::create_dir_all(&temp_dir)?;
 
+    // The default policy of reqwest Client supports max 10 attempts on HTTP redirect.
+    let client = Client::builder().redirect(Policy::default()).build()?;
+
+    // If input_xml exists, simply read it.
+    // If not, try to read from payload_url.
+    let response_text = match args.input_xml {
+        Some(name) => {
+            if name == "-".to_string() {
+                io::read_to_string(io::stdin())?
+            } else {
+                let file = File::open(name)?;
+                io::read_to_string(file)?
+            }
+        }
+        None => match args.payload_url {
+            Some(url) => fetch_url_to_string(&temp_dir, Url::try_from(url.as_str()).unwrap(), &client).await?,
+            None => return Err(format!("Either --input-xml or --payload-url must be given.").into()),
+        },
+    };
+
+    debug!("response_text: {:?}", response_text);
+
```

Comment:
We don't want to read the payload URL to a string and it won't even work because it's the binary payload to extract. It should either the `pkgs_to_dl` loop and instead directly invoke the download/verification methods or you can try to construct a fake package in `pkgs_to_dl` that holds the URL.

Reply:
Thanks, done.